pkg/api/handlers/compat/containers.go

```diff
@@ -85,6 +85,10 @@ func ListContainers(w http.ResponseWriter, r *http.Request) {
 		utils.InternalServerError(w, err)
 		return
 	}
+	// 0 means unlimited, need same with docker
+	if query.Limit == 0 {
+		query.Limit = -1
+	}
 	if _, found := r.URL.Query()["limit"]; found && query.Limit != -1 {
```

Comment:
Looking this over with more time, would it make more sense to change this line instead, to `&& query.Limit > 0` ?

Reply:
done